examples/albef/model.py

```diff
@@ -344,6 +348,155 @@ def forward(
+        itc_loss (ImageTextContrastiveLoss): Instantiated ImageTextContrastiveLoss.
+        itm_loss (ImageTextMatchingLoss): Instantiated ImageTextMatchingLoss.
+
+    Inputs:
+        image (Optional[Tensor] of shape (B, C, H, W)): Image features.
+            Required if is_train is True.
+            Required if input_type is "image" or "multimodal".
+        text (Optional[Tensor] of shape (B, L)): Text features.
+            Required if is_train is True.
+            Required if input_type is "text" or "multimodal".
+        text_atts (Tensor of shape (B, L)): Text attention mask.
+            Required if is_train is True.
+            Required if input_type is "text" or "multimodal".
+        idx (Tensor of shape (B)): Identifier for each image sample.
+            Required if is_train is True.
+        alpha (Optional[float]): The interpolation value between mlm_loss and loss_distill.
+            Default is 0.
+        input_type (Optional[str]): "image", "text", or "multimodal" indicating the encoding type.
+            Required if is_train is False.
+        is_train (Optional[bool]): Whether the model is in training.
+            Default is True.
+
+    Returns:
+        is_train is True:
+            Tensor: The sum of itc loss and itm loss.
+        is_train is False:
+            input_type is "image":
+                Tuple[Tensor, Tensor]: Image embeddings and projected image features.
+            input_type is "text":
+                Tuple[Tensor, Tensor]: Text embeddings and projected text features.
+            input_type is "multimodal"
+                Tensor: Scores for the retrieval task.
+    """
+
+    def __init__(
+        self,
+        model_with_similarity: ALBEFModelWithSimilarity,
+        itc_loss: ImageTextContrastiveLoss,
+        itm_loss: ImageTextMatchingLoss,
+    ) -> None:
+        super().__init__()
+        self.model_with_similarity = (
+            model_with_similarity  # TODO: rename model to model_with_similarity
+        )
+        self.itc_loss = itc_loss
+        self.itm_loss = itm_loss
+
+    def _train_forward(
+        self,
+        image: Tensor,
+        text: Tensor,
+        text_atts: Tensor,
+        idx: Tensor,
+        alpha: float,
+    ) -> Tensor:
+        encoder_output = self.model_with_similarity(image, text, text_atts, idx)
+
+        similarity_outputs = encoder_output.similarity
+        similarity_targets = encoder_output.sim_targets
+        itc_loss = self.itc_loss(
+            similarity_outputs.sim_i2t,
+            similarity_outputs.sim_t2i,
+            similarity_outputs.sim_i2t_m,
+            similarity_outputs.sim_t2i_m,
+            similarity_targets,
+            alpha,
+        )
+
+        pos_embeddings = encoder_output.multimodal_embeddings[:, 0, :]
+        neg_embeddings = encoder_output.multimodal_embeddings_neg[:, 0, :]
+        itm_loss = self.itm_loss(pos_embeddings, neg_embeddings)
+
+        loss = itc_loss + itm_loss
+        return loss
```

Comment:
Do we need a separate class to do this? Seems like we could just return the encoder_outputs and calculate the losses directly from those without bundling them with ALBEFModelWithSImilarity (at least for train, not sure how it will work for eval)